aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -538,19 +597,74 @@ def clear_context(self):
         self.context.clear()
 
 
-class MiddlewareHandler:
-    """Represents a callable Middleware function used to build a middleware stack when a route is resolved"""
+class MiddlewareStackWrapper:
```

Comment:
```suggestion
class MiddlewareFrame:
```

Still can't find a good name for this, `MiddlewareFrame` made slightly more sense in my head given the callable structure (current frame, previous frame, etc.).